### PR TITLE
Support valueFrom in extraEnv params

### PR DESCRIPTION
**Pull request by piratekev**
- Adds support for `extraEnv` parameters

Example configuration:
```
curity:
  admin:
    extraEnv:
      - name: AGENT_HOST
        valueFrom:
          fieldRef:
            fieldPath: status.hostIP

  runtime:
    extraEnv:
      - name: AGENT_HOST
        valueFrom:
          fieldRef:
            fieldPath: status.hostIP
```

### CODE REVIEW

File: idsvr/templates/deployment-runtime.yaml [hunk 1]
```diff
@@ -36,10 +36,15 @@ spec:
               value: {{ .Values.curity.healthCheckPort | quote }}
             - name: LOGGING_LEVEL
               value: {{ .Values.curity.runtime.logging.level }}
-            {{- range $env := .Values.curity.runtime.extraEnv }}
```
Comment (jlmart88) [inline idsvr/templates/deployment-runtime.yaml:39]: This should be runtime

File: idsvr/templates/deployment-admin.yaml [hunk 2]
```diff
@@ -52,8 +52,14 @@ spec:
...
             - name: {{ $env.name }}
+            {{- if $env.value }}
               value: {{ $env.value | quote }}
-          {{- end }}
+            {{- end }}
+            {{- if $env.valueFrom }}
+              valueFrom: 
+{{ $env.valueFrom | toYaml | trim | indent 16 }}
+            {{- end }}
+          {{- end}}
```
Comment (jlmart88) [inline idsvr/templates/deployment-admin.yaml:62]: `{{- end }}`

File: idsvr/templates/deployment-runtime.yaml [hunk 3]
```diff
@@ -36,10 +36,15 @@ spec:
...
+            {{- range $env := .Values.curity.admin.extraEnv }}
             - name: {{ $env.name }}
+              {{- if $env.value }}
               value: {{ $env.value | quote }}
-            {{- end }}
+              {{- end }}
```
Comment (jlmart88) [inline idsvr/templates/deployment-runtime.yaml:43]: this needs to be pulled out a level to close the range

File: idsvr/templates/deployment-admin.yaml [hunk 4]
```diff
@@ -52,8 +52,14 @@ spec:
           {{- end }}
           {{- range $env := .Values.curity.admin.extraEnv }}
             - name: {{ $env.name }}
+            {{- if $env.value }}
               value: {{ $env.value | quote }}
-          {{- end }}
+            {{- end }}
+            {{- if $env.valueFrom }}
+              valueFrom: 
+{{ $env.valueFrom | toYaml | trim | indent 16 }}
```
Comment (travisspencer) [inline idsvr/templates/deployment-admin.yaml:60]: looks like some formatting issues here

File: idsvr/templates/deployment-runtime.yaml [hunk 5]
```diff
@@ -38,7 +38,13 @@ spec:
               value: {{ .Values.curity.runtime.logging.level }}
             {{- range $env := .Values.curity.runtime.extraEnv }}
             - name: {{ $env.name }}
+              {{- if $env.value }}
               value: {{ $env.value | quote }}
+              {{- end }}
+              {{- if $env.valueFrom }}
+              valueFrom: 
+{{ $env.valueFrom | toYaml | trim | indent 16 }}
```
Comment (travisspencer) [inline idsvr/templates/deployment-runtime.yaml:46]: formatting here too

**Review by travisspencer — CHANGES_REQUESTED** [review]

**Review by anestos — APPROVED** [review]

File: idsvr/templates/deployment-admin.yaml (hunk 4 above)
Comment (anestos) [reply idsvr/templates/deployment-admin.yaml:60]: you can change to `{{- $env.valueFrom | toYaml | trim | indent 16 }}` and move this to the right below `valueFrom` to be more readable. This formatting is tecnhically still correct though

**Pull request merged**